zipline/src/commonMain/kotlin/app/cash/zipline/internal/bridge/outbound.kt

```diff
@@ -98,17 +100,19 @@ internal class OutboundCall(
 
   @PublishedApi
   internal suspend fun <R> invokeSuspending(serializer: KSerializer<R>): R {
+    require(callCount++ == parameterCount)
     return suspendCoroutine { continuation ->
-      require(callCount++ == parameterCount)
-      val callbackName = endpoint.generateName()
-      val callback = RealSuspendCallback(callbackName, continuation, serializer)
-      endpoint.set<SuspendCallback>(callbackName, EmptySerializersModule, callback)
-      endpoint.outboundChannel.invokeSuspending(
-        instanceName,
-        funName,
-        arguments.toTypedArray(),
-        callbackName
-      )
+      CoroutineScope(context.endpoint.dispatcher).launch {
```

Comment:
I'm thinking that you're looking for this pattern: https://github.com/square/retrofit/blob/6c34d7b6b534325408948d37c573e1fe40154a2a/retrofit/src/main/java/retrofit2/KotlinExtensions.kt#L113-L114

A naked `CoroutineScope` is a huge red flag for me.

Reply:
Agreed! I’d like some help with this! I tried the above, it didn’t work:

```
java.lang.ClassCastException: class kotlin.coroutines.jvm.internal.CompletedContinuation cannot be cast to class kotlinx.coroutines.internal.DispatchedContinuation (kotlin.coroutines.jvm.internal.CompletedContinuation and kotlinx.coroutines.internal.DispatchedContinuation are in unnamed module of loader 'app')
	at kotlinx.coroutines.CoroutineDispatcher.releaseInterceptedContinuation(CoroutineDispatcher.kt:107)
	at kotlin.coroutines.jvm.internal.ContinuationImpl.releaseIntercepted(ContinuationImpl.kt:118)
	at kotlin.coroutines.jvm.internal.BaseContinuationImpl.resumeWith(ContinuationImpl.kt:39)
	at app.cash.zipline.internal.bridge.OutboundCall$RealSuspendCallback.call(outbound.kt:129)
	at app.cash.zipline.internal.bridge.OutboundCall$invokeSuspending$2$1$run$1$create$1.call(outbound.kt)
```

Reply:
Well to be honest the Retrofit code is stolen from Roman so... we can figure it out the right thing eventually.